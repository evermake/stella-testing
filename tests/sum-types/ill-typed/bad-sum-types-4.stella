language core;
extend with #sum-types;

fn g(x : Nat + (Bool + (fn(Nat) -> Nat))) -> Nat {
  return match x {
      inl(n) => succ(n)
  }
}

fn main(x : Nat) -> Nat {
  return g(inr(inr(fn(n : Nat) { return g(inl(n)) })))
}
