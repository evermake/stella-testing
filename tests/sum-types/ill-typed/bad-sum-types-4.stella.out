An error occurred during typechecking!
Type Error Tag: [ERROR_ILLEGAL_MATCHING_FOR_TYPE]
pattern matching on functions is not allowed